cmake/EthCompilerSettings.cmake

```diff
@@ -14,18 +14,15 @@
 #
 # These settings then end up spanning all POSIX platforms (Linux, OS X, BSD, etc)
 
-include(CheckCXXCompilerFlag)
+include(EthCheckCXXCompilerFlag)
 
-check_cxx_compiler_flag(-fstack-protector-strong have_stack_protector_strong)
-if (have_stack_protector_strong)
-	add_compile_options(-fstack-protector-strong)
-else()
-	check_cxx_compiler_flag(-fstack-protector have_stack_protector)
-	if(have_stack_protector)
-		add_compile_options(-fstack-protector)
-	endif()
+eth_add_cxx_compiler_flag_if_supported(-fstack-protector-strong have_flag)
+if(NOT have_flag)
+	eth_add_cxx_compiler_flag_if_supported(-fstack-protector)
```

Comment:
Do you want to create a PR for the cmake and the stack protector change (and the URL stuff above)? That is something we should merge now.

Reply:
"stack protector" is only a refactor (I created a helper function to handle flags). But you cannot merge "fallthough warning" as solidity will not compile.

Reply:
I only meant the stack protector part and the helper to be merged.